sold.cc

```diff
@@ -737,12 +736,53 @@ std::vector<std::string> Sold::GetLibraryPaths(const ELFBinary* binary) {
     return library_paths;
 }
 
-void Sold::ResolveLibraryPaths(const ELFBinary* root_binary) {
+std::vector<ELFBinary*> ToplologicalSort(std::map<std::string, ELFBinary*> needed_to_bin) {
+    std::vector<ELFBinary*> ret;
+    std::queue<std::string> next_node;
+    std::map<std::string, int> n_incoming_edge;
+
+    for (const auto& p : needed_to_bin) n_incoming_edge[p.first] = 0;
+    for (const auto& p : needed_to_bin) {
+        for (const std::string& s : p.second->neededs()) {
+            if (needed_to_bin.count(s)) n_incoming_edge[s]++;
+        }
+    }
+
+    for (const std::pair<std::string, int> p : n_incoming_edge) {
```

Comment:
Is this order compatible with glibc? I doubt alphabetical order is not the tie breaker for leaf nodes. I guess any order is OK in terms of ELF standard as long as they are topologically ordered. However, I think it's better to try sold to be compatible with glibc as people may depend on the behavior of glibc.